bin/install

```diff
@@ -1,158 +1,449 @@
+    command mkdir -p "$FISH_CONFIG/conf.d"
+    generate_bootstrap "$OMF_CONFIG" > "$vendor_config_file"
+
+  # If the user doesn't have their own config file, we'll just use that as our bootstrap.
+  else if not test -e "$fish_config_file"
+    say "Writing bootstrap to $fish_config_file..."
+    generate_bootstrap "$OMF_CONFIG" > "$fish_config_file"
 
-  report progress "Adding startup code to fish config file..."
+  # Even though config.fish already exists, we can prepend to it if the user is OK with it.
+  else if confirm "Would you like Oh My Fish to be added to your configuration automatically?"
+    say "Prepending bootstrap to $fish_config_file..."
 
-  set template "templates/config.fish"
-  set replacements "s|{{OMF_CONFIG}}|$OMF_CONFIG|"
+    # Create a temporary file to store the combined config so that we can write atomically.
+    generate_bootstrap "$OMF_CONFIG" | command cat - "$fish_config_file" > "$fish_config_file.tmp"
+      or abort "Error prepending config file"
 
-  if test "$OMF_CONFIG" != "$XDG_CONFIG_HOME/omf"
-    set replacements "$replacements;s|#set|set|"
+    # Swap in the prepended file.
+    command mv "$fish_config_file.tmp" "$fish_config_file"
+      or abort "Error moving file to $fish_config_file"
+
+  # We don't have any options left, so let the user set up the bootstrap manually.
+  else
+    say "For Oh My Fish to work properly, please paste the code below into a file that Fish can run at startup:"
+    echo
+    generate_bootstrap
   end
 
-  sed  "$replacements" "$OMF_PATH/$template" > "$fish_config_file"
+  # Backup the user's theme settings. This can be removed when OMF no longer touches this file to switch themes.
+  backup_file "$FISH_CONFIG/functions/fish_prompt.fish"
 
-  report progress "Building Oh My Fish configuration..."
+  say "Building Oh My Fish configuration..."
 
+  # Set up the Oh My Fish configuration directory.
   if not test -d "$OMF_CONFIG"
-    mkdir -p "$OMF_CONFIG"
+    command mkdir -p "$OMF_CONFIG"
   end
 
-  test -f "$OMF_CONFIG/bundle"; or echo "theme default" > "$OMF_CONFIG/bundle"
-  test -f "$OMF_CONFIG/theme";  or echo "default" > "$OMF_CONFIG/theme"
+  test -f "$OMF_CONFIG/bundle";
+    or echo "theme default" > "$OMF_CONFIG/bundle"
+  test -f "$OMF_CONFIG/theme"
+    or echo "default" > "$OMF_CONFIG/theme"
 
+  # Install plugins
   fish -c "omf install"
+    or abort "Error installing plugins"
 end
 
-function main
-  report progress "Installing Oh My Fish to $OMF_PATH..."
 
-  if test -d "$OMF_PATH"
-    report error "Aborting: Existing installation detected."
-  end
+# Generates the bootstrap code used to initialize Oh My Fish on shell startup.
+function generate_bootstrap -a config_path
+echo '# Path to Oh My Fish install.
+set -q XDG_DATA_HOME
+  and set -gx OMF_PATH "$XDG_DATA_HOME/omf"
+  or set -gx OMF_PATH "$HOME/.local/share/omf"
+
+# Customize Oh My Fish configuration path.
+#set -gx OMF_CONFIG "'$config_path'"
```

Comment:
It isn't double-quoting; it's printing a double quote, exiting the string, printing `$config_path`, opening the string again, and printing another double quote.